core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java

```diff
@@ -670,6 +672,17 @@ public void checkCanExecuteFunction(SecurityContext context, String functionName
         }
     }
 
+    @Override
+    public void checkCanExecuteTableProcedure(SecurityContext context, QualifiedObjectName table, String procedure)
+    {
+        if (shouldDenyPrivilege(context.getIdentity().getUser(), table + "." + procedure, EXECUTE_TABLE_PROCEDURE)) {
+            denyExecuteTableProcedure(table.toString(), procedure);
+        }
+        if (denyPrivileges.isEmpty()) {
```

Comment:
why this if?

Reply:
I am dumbly following the pattern in this class.